src/observer.ts

```diff
@@ -417,13 +460,37 @@ export class DataLayerObserver {
     // rule properties override global ones
     const readOnLoad = ruleReadOnLoad === undefined ? globalReadOnLoad : ruleReadOnLoad;
 
-    if (!source || !destination) {
+    if (!source) {
       Logger.getInstance().error(LogMessageType.RuleInvalid,
-        { rule: id, source, reason: `Missing ${source ? 'destination' : 'source'}` });
+        { rule: id, source, reason: 'Missing source' });
       Telemetry.error(errorType.invalidRuleError);
       return;
     }
 
+    // sanity check destination and fsApi parameters
+    if (!destination && !fsApi) {
+      Logger.getInstance().error(LogMessageType.OperatorError, {
+        reason: LogMessage.MissingDestination,
+      });
+      Telemetry.error(errorType.operatorError);
+      return;
+    }
+
+    if (destination && fsApi) {
```

Comment:
While conceptually incorrect (competing destinations), I wonder if we should prefer fsApi if it is also used with destination. In this scenario, we still still send data versus nothing.  We could emit an error to indicate the problem, and the user would be able to see the event in the replay along with the indication there are competing destinations.